### ISSUE TITLE
Testing new releases

### ISSUE THREAD

**Issue by williamparry** [williamparry/cloudRIG#73]
Looking for people able to test new releases. I don't have a mac, so someone with a mac would be handy.

**Comment by williamparry** [comment]
https://www.bountysource.com/issues/55762280-testing-new-releases

**Comment by williamparry** [comment]
Thanks for that - could you paste them into your comment rather than zip? It's useful for future diagnostics (and other people who might stumble here with a related issue one day).

Are you running the [build script](https://github.com/williamparry/cloudRIG/blob/dev/gui/package.json#L7)? You should be able to run it and it compile an executable for you to use.

**Windows:** You need to change it to ```set BROWSER=none``` with the additional ```set```. Then you'll probably come across this [Electron issue](https://github.com/atom/atom/issues/14227), so hold off on that for now.

**Ubuntu:** Not sure about the empty UI issue, but don't worry about the Check for Update issue, I don't think it is a real problem.

**Mac:** This should be working, at least. The problem I'm facing at the moment is that for some reason the modal dialog is half off screen (is fine on previous release). Let's get that fixed first.

Another thing, do you have a separate AWS account for testing? If you follow [these instructions](https://github.com/williamparry/cloudRIG/wiki/AWS-Testing) I'll add you to my testing group so it doesn't cost you anything to run tests.

**Comment by williamparry** [comment]
**Update:** if you're running ```npm run dev``` it runs the electron and react projects at the same time, which means if the electron process finishes first, there will be a blank webpage (since the react dev server is still starting up). Just refresh.

**Comment by artem7902** [comment]
Ok, i use build script but have the same problems. For Windows after adding set i have Electron issue. About test account, i have some problems with activation new AWS account in my country, automatically call doesn't work. Why you can't just create new iam user for me? 

Ubuntu and MacOS:

1. I have this screen after start. Sport request active, instance running 30 minutes. It fixed after restart gui.
![image](https://user-images.githubusercontent.com/26010756/37253063-2f57f22e-2535-11e8-8689-2146dffac30f.png)
2. If i close electron app using button with cross in the right top corner and then start it again i have this message. Fixed after manually kill process.
![image](https://user-images.githubusercontent.com/26010756/37253081-62d2deb6-2535-11e8-9980-895e58bf2b29.png)
3. Always have this error in console after start server:
> Checking for update
Error: HttpError: 404 Not Found
"method: undefined url: https://github.com/williamparry/cloudRIG/releases/download/v0.0.1-alpha.2/latest-linux.yml\n\nPlease double check that your authentication token is correct. Due to security reasons actual status maybe not reported, but 404.\n"

4. Stoppen also doesn't works, i only have log message in bottom after each click on stop button. Fixed only after manually stop from aws console.
![image](https://user-images.githubusercontent.com/26010756/37253103-8ca88ed4-2535-11e8-8539-93d584f770d6.png)

**Comment by williamparry** [comment]
**1.** If you're running the dev command the order of build might mean the electron app appears before react has finished, so you'll need to refresh.

**2.** Yeah, that's annoying. You can run each process independently in different terminals to get around that.

**3.** Not a bug, don't worry

**4.** SSM was broken until I fixed it just now. Please get the latest version.

Re: IAM. Part of the reason to use a separate account is that it puts all of your AWS usage under cloudRIG's billing (if you have other projects on AWS, I will be charged for that also). If cloudRIG is all you're doing under your account, then no probs, I can make a user for you.

**Comment by artem7902** [comment]
1. Now it working well
4. Now instance terminated but spot request standing and create a new instance after first terminated.

I will be happy if you create new IAM user for me, I will use it only for cloudRIG testing.

**Comment by williamparry** [comment]
Is that the yandex.ru account on your Github profile? (I need to associate the IAM user with your AWS account).

I fixed some things yesterday, so you may want to delete the lambdas, delete the IAM roles and get it to regenerate. In the CLI there is an advanced menu to help with it.

**Comment by artem7902** [comment]
please use nataef80@gmail.com for IAM

**Comment by williamparry** [comment]
Just invited. Did you get an email?

**Comment by artem7902** [comment]
Yes, I accepted it.

**Comment by williamparry** [comment]
OK great. Please remember to turn off the instances when you're done :)

**Comment by artem7902** [comment]
Ok. After cleaning everything works well.

**Comment by williamparry** [comment]
@artem7902 does the latest version run on Ubuntu? I'm keen to release for other platforms.

**Comment by artem7902** [comment]
@williamparry Hi, I have some issues with Ubuntu: 
1. After clicking "Open VNC" button nothing happens. As I understood the reason in "opn" library,  "open" library worked for me before. Stack trace:
2. I got next error when tried schedule shutdown
![1](https://user-images.githubusercontent.com/26010756/40741560-878b108a-6454-11e8-9d39-e364d0fde5b5.png)
3. I fixed the second issue but UI doesn't show schedule shutdown state. There are no remaining time and "Cancel schedule stop" button. Also, scheduling stop doesn't work, it doesn't terminate the instance.
![image](https://user-images.githubusercontent.com/26010756/40744986-e8369170-645e-11e8-9ffc-8dda330ee319.png)

**Comment by williamparry** [comment]
Yeah I think that error is around the SSM check for scheduled tasks. It's related to here: https://github.com/williamparry/cloudRIG/issues/81 if you want to take a look at that?

**Comment by artem7902** [comment]
@williamparry Ok, I can take this issue few later, some kind of busy now :)

**Comment by williamparry** [comment]
I've put the fix in dev.

**Comment by pjz** [comment]
As part of testing updates, maybe set up something like https://greenkeeper.io/ ; the nodejs ecosystem moves pretty quickly still.
